python/segyio/tools.py

```diff
@@ -403,3 +407,429 @@ def resample(f, rate = None, delay = None, micro = False,
     f._samples = (np.arange(len(f.samples)) * rate) + t0
 
     return f
+
+
+def from_array(filename, data, iline=189,
+                               xline=193,
+                               sample_format=SegySampleFormat.IBM_FLOAT_4_BYTE,
+                               dt=4000,
+                               delay_recording_time=0):
+    """ Create a new SEGY file from an n-dimentional array
+
+    Create an structured SEGY file with defaulted headers from a 2-, 3- or
+    4-dimensional array. from_array() recognizes the shape of the array and
+    calls the correct from_arrayxD() function. I.e. calling this with a
+    2-dimensional array produces the exact same result as calling from_array2D()
+    directly.
+
+    Parameters
+    ----------
+
+    filename : string-like
+        Path to new file
+
+    data : 2-,3- or 4-dimensional array-like
+
+    iline : int or segyio.TraceField
+        Inline number field in the trace headers. Defaults to 189 as per the
+        SEG-Y rev1 specification
+
+    xline : int or segyio.TraceField
+        Crossline number field in the trace headers. Defaults to 193 as per the
+        SEG-Y rev1 specification
+
+    sample_format : int or segyio.SegySampleFormat
+        Sample format field in the trace header. Defaults to IBM float 4 byte
+
+    dt : int-like
+        sample interval
+
+    delay_recording_time : int-like
+
+    Notes
+    -----
+
+    .. versionadded:: 1.8
+
+    Examples
+    --------
+
+    Create a file from a 3D array, open it and read an iline:
+
+    >>> segyio.tools.from_array(path, array3d)
+    >>> segyio.open(path, mode) as f:
+    ...     iline = f.iline[0]
+    ...
+    """
+
+    data = np.asarray(data)
+    dimentions = len(data.shape)
+
+    if dimentions not in range(2,5):
+        problem = "Expected 2, 3 or 4 dimensions, {} was given".format(dimentions)
+        raise ValueError(problem)
+
+    if dimentions is 2:
```

Comment:
While cute and probably ok, you actually want `==` here, since integer-constants probably won't be the same object as an int-var